src/formatting.ts

```diff
@@ -45,6 +46,74 @@ function toggleCodeBlock() {
     return editor.insertSnippet(new SnippetString('```$0\n$TM_SELECTED_TEXT\n```'));
 }
 
+enum QuoteState {
+    // State 1: part of lines has been quoted, need to be quoted
+    PARTIALQUOTED,
+    // State 2: all of lines has been quoted, need to be unquoted
+    FULLQUOTED,
+}
+
+function getQuoteState(lineArray: Array<string>): QuoteState {
+    let cntQuoted = 0;
+    console.log("array:", lineArray)
+    lineArray.forEach(function(line) {
+        if (line.startsWith(">")) {
+            cntQuoted += 1;
+        }
+    })
+
+    return cntQuoted != lineArray.length ? QuoteState.PARTIALQUOTED : QuoteState.FULLQUOTED;
+}
+
+function quoteLine(line: string): string {
+    if(!line.startsWith("> ") && line.startsWith(">")) {
+        return line.replace(/>/g, "> ");
+    } else if(!line.startsWith("> ")){
+        return "> " + line;
+    } else {
+        return line;
+    }
+}
+
+function unquoteLine(line: string): string {
+    if (line.startsWith("> ")) {
+        return line.slice(2);
+    } else {
```

Comment:
`line.startsWith('>')`?

Reply:
Yes, you're right. I thought adding a space is more beautiful and standard. But actually no such setting in [github markdown](https://docs.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax#ignoring-markdown-formatting).I will modify this.

Reply:
Well, I think we can definitely advocate adding a space after `>` (actually I like it). I meant we still need to handle the case if there isn't a space.

Reply:
Ok～I will modify the space back and skip the line which it hasn't appended space.